### PR TITLE
[FEATURE] add composer_script task

### PR DESCRIPTION
**Pull request by Kanti**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | yes
| Fixed tickets | #212 

# New Task Checklist:

- [x] Is the README.md file updated?
- [x] Are the dependencies added to the composer.json suggestions?
- [x] Is the doc/tasks.md file updated?
- [x] Are the task parameters documented?
- [x] Is the task registered in the tasks.yml file?
- [x] Does the task contains phpspec tests?
- [x] Is the configuration having logical allowed types?
- [x] Does the task run in the correct context?
- [x] Is the `run()` method readable?
- [x] Is the `run()` method using the configuration correctly?
- [x] Are all CI services returning green?

### CODE REVIEW

**Comment by Kanti** [comment]
Please Review! 😅

**Comment by veewee** [comment]
Also thanks for this PR!
I've got the same comments as on the npm script task:

- Can you set default triggered_by extensions for the composer script? I was thinking about [php, phtml] (Note: Currently the check for run on every extension is not added?)
- Can you specify an alternative composer.json location just like in the composer task?

**Comment by Kanti** [comment]
added changes Please Review 😅

File: src/GrumPHP/Task/ComposerScript.php [hunk 1]
```diff
@@ -62,6 +64,7 @@ public function run(ContextInterface $context)
         $arguments->addRequiredArgument('%s', $config['script']);
 
         $process = $this->processBuilder->buildProcess($arguments);
+        $process->setWorkingDirectory($config['working_directory']);
```
Comment (veewee) [inline src/GrumPHP/Task/ComposerScript.php]: The run-script command actually has a working dir option.
Maybe it's better to use that one if the working_directory is set and default it to null?

```
  -d, --working-dir=WORKING-DIR  If specified, use the given directory as working directory.
```

**Comment by Kanti** [comment]
now composer_script uses the -d option from composer.

**Comment by veewee** [comment]
Thanks!

**Pull request merged**